cmake/alpakaCommon.cmake

```diff
@@ -261,278 +261,146 @@ endif()
 #-------------------------------------------------------------------------------
 # Find CUDA.
 if(ALPAKA_ACC_GPU_CUDA_ENABLE)
+    include(CheckLanguage)
+    check_language(CUDA)
 
-    if(NOT DEFINED ALPAKA_CUDA_VERSION)
-        set(ALPAKA_CUDA_VERSION 9.0)
-    endif()
```

Comment:
Maybe I missed this, but what happens if I do not set `ALPAKA_CUDA_VERSION`?

Also see subsequent comment.

Reply:
I am currently experimenting with getting rid of this completely. The new way of doing this seems to set `CUDACXX` env variable to the CUDA compiler of your choice and else CMake will select one automatically.

Reply:
I hopefully got rid of the variable now.
If you want to select the CUDA compiler explicitly, you can specify `CMAKE_CUDA_COMPILER: nvcc` of `CMAKE_CUDA_COMPILER: clang`. The default seems to be `nvcc`. If you want to select a specific version different from the default that CMake takes, you have to specify `CUDACXX`.
If you selected `CMAKE_CUDA_COMPILER: clang` and want to specify a specific CUDA toolkit different from the default that CMake takes, you have to specify `CUDAToolkit_ROOT`.

Reply:
Thank you for the elaborated reply! The new approach also seems more reasonable to me, since I usually want to select a specific compiler instead of a CUDA version (with the selected compiler entailing a CUDA version).